### ISSUE TITLE
Can't deploy my code on AWS Lambda using CodeCommit auto push I keep get this error: `npm ERR! Failed at the prisma@4.14.1 preinstall script.`

### ISSUE THREAD

**Issue by adelynx** [prisma/prisma#19398]
### Bug description

When I push my commits to CodeCommit it automatically deploys it to AWS Lambda application, but I am getting this error all the time:

```
2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:16 Waiting for agent ping
-- | --
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:17 Waiting for DOWNLOAD_SOURCE
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:18 Phase is DOWNLOAD_SOURCE
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:18 CODEBUILD_SRC_DIR=/codebuild/output/src951447098/src
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:18 YAML location is /codebuild/output/src951447098/src/buildspec.yml
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:18 Setting HTTP client timeout to higher timeout for S3 source
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:18 Processing environment variables
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Moving to directory /codebuild/output/src951447098/src
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Registering with agent
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Phases found in YAML: 3
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 BUILD: 1 commands
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 INSTALL: 1 commands
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 PRE_BUILD: 1 commands
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Phase complete: DOWNLOAD_SOURCE State: SUCCEEDED
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Phase context status code: Message:
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Entering phase INSTALL
  | 2023-05-23T12:33:24.368+01:00 | [Container] 2023/05/23 11:33:20 Running command npm install
  | 2023-05-23T12:33:48.419+01:00 | npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
  | 2023-05-23T12:34:05.216+01:00 |  
  | 2023-05-23T12:34:05.216+01:00 | > prisma@4.14.1 preinstall /codebuild/output/src951447098/src/node_modules/prisma
  | 2023-05-23T12:34:05.216+01:00 | > node scripts/preinstall-entry.js
  | 2023-05-23T12:34:05.216+01:00 |  
  | 2023-05-23T12:34:05.216+01:00 | ┌─┐
  | 2023-05-23T12:34:05.216+01:00 | │ …│
  | 2023-05-23T12:34:05.216+01:00 | └┘
  | 2023-05-23T12:34:05.216+01:00 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/chokidar/node_modules/fsevents):
  | 2023-05-23T12:34:05.216+01:00 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @nestjs/common@9.4.2 requires a peer of cache-manager@<=5 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @nestjs/core@9.4.2 requires a peer of @nestjs/microservices@^9.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @nestjs/core@9.4.2 requires a peer of @nestjs/websockets@^9.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @nestjs/swagger@6.3.0 requires a peer of @fastify/static@^6.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @aws-sdk/signature-v4-multi-region@3.337.0 requires a peer of @aws-sdk/signature-v4-crt@^3.118.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @aws-sdk/util-user-agent-node@3.337.0 requires a peer of aws-crt@>=1.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN node-fetch@2.6.11 requires a peer of encoding@^0.1.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @nestjs/testing@9.4.2 requires a peer of @nestjs/microservices@^9.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN jest@29.5.0 requires a peer of node-notifier@^8.0.1 \|\| ^9.0.0 \|\| ^10.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN jest-cli@29.5.0 requires a peer of node-notifier@^8.0.1 \|\| ^9.0.0 \|\| ^10.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN ts-node@10.9.1 requires a peer of @swc/core@>=1.2.50 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN ts-node@10.9.1 requires a peer of @swc/wasm@>=1.2.50 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @jest/core@29.5.0 requires a peer of node-notifier@^8.0.1 \|\| ^9.0.0 \|\| ^10.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 | npm WARN @jest/reporters@29.5.0 requires a peer of node-notifier@^8.0.1 \|\| ^9.0.0 \|\| ^10.0.0 but none is installed. You must install peer dependencies yourself.
  | 2023-05-23T12:34:05.216+01:00 |  
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! code ELIFECYCLE
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! errno 1
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! prisma@4.14.1 preinstall: `node scripts/preinstall-entry.js`
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! Exit status 1
  | 2023-05-23T12:34:05.216+01:00 | npm ERR!
  | 2023-05-23T12:34:05.216+01:00Copynpm ERR! Failed at the prisma@4.14.1 preinstall script. | npm ERR! Failed at the prisma@4.14.1 preinstall script.
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  | 2023-05-23T12:34:05.216+01:00 |  
  | 2023-05-23T12:34:05.216+01:00 | npm ERR! A complete log of this run can be found in:
  | 2023-05-23T12:34:05.216+01:00Copynpm ERR!     /root/.npm/_logs/2023-05-23T11_34_03_136Z-debug.log | npm ERR! /root/.npm/_logs/2023-05-23T11_34_03_136Z-debug.log
  | 2023-05-23T12:34:05.216+01:00 |  
  | 2023-05-23T12:34:05.216+01:00 | [Container] 2023/05/23 11:34:03 Command did not exit successfully npm install exit status 1
  | 2023-05-23T12:34:05.216+01:00 | [Container] 2023/05/23 11:34:03 Phase complete: INSTALL State: FAILED
  | 2023-05-23T12:34:05.216+01:00 | [Container] 2023/05/23 11:34:03 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: npm install. Reason: exit status 1
```

### How to reproduce

When I push my commits to CodeCommit it automatically deploy it to AWS Lambda application.

This is my `buildspec.yml` so far. It's failing on `npm install`

```yml
version: 0.2
phases:
  install:
    commands:
      # Install all dependencies (including dependencies for running tests)
      - npm install
  pre_build:
    commands:
      # Remove all dependencies not needed for the Lambda deployment package (the packages from devDependencies in package.json)
      - npm prune --production
  build:
    commands:
      # Use AWS SAM to package the application by using AWS CloudFormation
      - aws cloudformation package --template template.yml --s3-bucket $S3_BUCKET --output-template-file template-export.yml
artifacts:
  files:
    - template-export.yml
```

### Expected behavior

I am expecting to be able to install dependencies and get deployed and ran as it should.

### Prisma information

```
generator client {
  provider      = "prisma-client-js"
  // binaryTargets = [env("PRISMA_BINARY_TARGET")]
  binaryTargets = ["rhel-openssl-1.0.x"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}
```

### Environment & setup

- OS: AWS Lambda
- Database: MongoDB Atlas Cloud
- Node.js version: nodejs18.x

### Prisma Version

```
4.14.1
```

**Comment by DirectStack** [comment]
I always exe  ```pnpm i --ignore-scripts``` when deploying. May this help you

**Comment by janpio** [comment]
Can you make sure to provide all information we would need to reproduce this ourselves @adelynx? Assume we know nothing about CodeCommit and have to figure things out as we go - so any information is useful. Thanks.

**Comment by adelynx** [comment]
@janpio After a couple of tries and debugging I found the issue with the `rhel-openssl` version. Basically, I set up a CodeCommit with CodeBuild and CodeDeploy to auto-deploy my app when push a commit. It turns out that the `rhel-openssl` version at the CodeBuild level is not the same as the one in CodeBuild and AWS Lambda so the Prisma client has been building with a version and tried to run it in an environment with a different version that's what was causing the issue. It took me a bit of time to figure it out. I can confirm that there is no issue with Prisma.

Cheers,
Adel
